ecs-agent/api/resource/resource_attachment.go

```diff
@@ -174,6 +176,22 @@ func (ra *ResourceAttachment) SetSentStatus() {
 	ra.AttachStatusSent = true
 }
 
+// IsAttached checks if the resource attachment has been found attached on the host
+func (ra *ResourceAttachment) IsAttached() bool {
+	ra.guard.RLock()
+	defer ra.guard.RUnlock()
+
+	return ra.Status == status.AttachmentAttached
```

Comment:
Good question. It should be thread safe here since we're locking the read access for the resource attachment object and then unlocking it at the end (unless I am missing!). This should behave similar to `AttachStatusSent`.

Reply:
I meant the naming of `Status` here. The overall setter is thread safe but the field itself is not (other wise we are not protecting it with locks). Should we make it a private field, or change its naming with "Unsafe" in it?

Reply:
Synced offline, this is not a trivial change and will be done as a follow up

Reply:
Thank you!